src/v/archival/service.cc

```diff
@@ -234,6 +234,7 @@ ss::future<> scheduler_service_impl::add_ntp_archiver(
     return archiver->download_manifest().then(
       [this, archiver](cloud_storage::download_result result) {
           auto ntp = archiver->get_ntp();
+          auto part = _partition_manager.local().get(ntp);
```

Comment:
should we check if partition is present ?

Reply:
`archiver` can only be created if partition exists. But there's a scheduling point between the archiver creation and the check. I think it can be solved by adding check to archiver or by making archiver decide what background operations need to be started.